pkg/icingadb/history/retention.go

```diff
@@ -172,6 +173,20 @@ func (r *Retention) Start(ctx context.Context) error {
 			r.logger.Debugf("Skipping history retention for category %s", stmt.Category)
 			continue
 		}
+		// #nosec G115 -- First if-clause checks int overflow
+		if days > math.MaxInt64 || time.Now().AddDate(0, 0, -int(days)).Before(time.Unix(0, 0)) {
```

Comment:
> but after plugging `MaxUint16` into my playbook example, I'm confused as it says 1830 and I'd have expected something like 1845, is it an overflow or some weird calendar thing?

D'oh! I forget that `time.Now()` is always 2009 in the Go playground...

Reply:
I wouldn't care too much about 32bit builds but reducing days to `uint16` sounds good to me too. Though I would do that in a separate PR.

Reply:
> I wouldn't care too much about 32bit builds

To fix that part, all the change that's needed would be to use `math.MaxInt` instead of `math.MaxInt64`, I wouldn't say that's too much. But that part will likely become irrelevant anyway due to `time.Time` overflowing.

Reply:
Changed for int32 and did some further testing: https://go.dev/play/p/92-1ROGJBOl